crates/anvil/core/src/eth/transaction/mod.rs

```diff
@@ -1206,31 +1112,122 @@ pub enum TypedReceipt<T = Receipt<alloy_primitives::Log>> {
     #[serde(rename = "0x4", alias = "0x04")]
     EIP7702(ReceiptWithBloom<T>),
     #[serde(rename = "0x7E", alias = "0x7e")]
-    Deposit(DepositReceipt<T>),
+    Deposit(OpDepositReceiptWithBloom),
```

Comment:
this is a type alias for `ReceiptWithBloom<OpDepositReceipt<T>>`

imo, in this case, we either make `T` the generic for the log type, or we just ditch `T` entirely and make the other `ReceiptWithBloom<T>` concrete: `ReceiptWithBloom<Receipt<Log>>`